### PR TITLE
Enhance "Copy for Context" Extension with New Copy Commands and Tab Management Features

### PR DESCRIPTION
**Pull request by radleta**
This PR introduces several new features and improvements to the "Copy for Context" VS Code extension:

### New Features:
- **Copy This File for Context:** Allows users to copy the currently active file as formatted Markdown.
- **Copy This Tab Group for Context:** Adds the ability to copy all files within the same tab group as formatted Markdown.
- **Copy All Open Tabs for Context:** Enables users to copy all open tabs as formatted Markdown.

### Improvements:
- **Context Menu Organization:** 
  - Grouped related copy commands together in the editor title context menu for better organization.
  - Added a horizontal separator to distinguish these commands from other options.
- **Documentation Updates:** 
  - The README and CHANGELOG have been updated to reflect the new commands and features.

These changes enhance the extension's usability, particularly for users who manage their files across multiple tabs and tab groups.

Please review the changes and merge if everything looks good.

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -1,56 +1,143 @@
-import * as vscode from 'vscode';
-import * as path from 'path';
-import * as fs from 'fs';
+import * as vscode from "vscode";
```
Comment (collinc777) [inline src/extension.ts:1]: The different formatter is causing a an expanded diff but that's on me

File: package.json [hunk 2]
```diff
@@ -41,7 +89,8 @@
     "pretest": "npm run compile && npm run lint",
     "lint": "eslint src --ext ts",
     "test": "node ./out/test/runTest.js",
-    "package": "vsce package"
+    "package": "vsce package",
+    "cpi": "npm run compile && npm run package && code-insiders --install-extension $(ls context-copy-*.vsix | head -n 1)"
```
Comment (collinc777) [inline package.json:93]: Why this?

**Review by collinc777 — COMMENTED** [review]
Thanks for the PR! The Diff is a little muddy.

We definitley need a test suite now that people depend on this.

I'm not going to block on the formatting diff but if you want to clean it up go for it.

Before I merge, I need to know a little bit about how you tested this. Can you provide that info please?

File: src/extension.ts (hunk 1 above)
Comment (radleta) [reply src/extension.ts:1]: Anything you want me to do on this?

File: package.json (hunk 2 above)
Comment (radleta) [reply package.json:93]: Shorthand for quickly compiling, packaging, then installing on my local to test the extension out in vscode insiders. I can remove it though and set up a private alias should you want me to do so.

**Comment by radleta** [comment]
> Thanks for the PR! The Diff is a little muddy.
> 
> We definitley need a test suite now that people depend on this.
> 
> I'm not going to block on the formatting diff but if you want to clean it up go for it.
> 
> Before I merge, I need to know a little bit about how you tested this. Can you provide that info please?

Thanks for the feedback. I’ve been testing the new version manually while using it daily, and so far, it’s been solid. I did find an issue with the BOM getting included, causing a corrupt character in the output.

Do you have any specific expectations for testing beyond what I’ve been doing? Also, let me know if you want me to clean up the formatting diff before you merge.

Richard

**Comment by radleta** [comment]
Just following up on the formatting diff. Would you like me to make any specific changes? If so, how would you go about cleaning it up?

**Comment by collinc777** [comment]
I just updated the package to include formatting stuff for yah. If you could resolve the mere conflicts and run prettier then we should be good to go

**Comment by radleta** [comment]
Merge resolved and prettier done. Please wait on me to test. I need some time to test these changes out. I'll drop a note when I feel the changes are solid.

**Comment by radleta** [comment]
@collinc777 I did some testing and everything seems to check out. 👍

**Comment by radleta** [comment]
@collinc777 Thanks for the merge! Did my prettier not match what you put in? Asking so I can ensure I check things to try to figure out why it didn't properly follow repo settings.

**Comment by collinc777** [comment]
yeah for w/e reason I had to re-run it. No big deal

**Pull request merged**